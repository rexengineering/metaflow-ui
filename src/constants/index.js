export const BUSY = "BUSY";
export const OFFLINE = "OFFLINE";
export const ONLINE = "ONLINE";

export const ActionItemDeleteType = "ActionDeleteType";
export const ActionItemCheckType = "ActionCheckType";
export const ActionItemDisplayOnlyType = "ActionDisplayOnlyType";

export const InitialButtonState = 'initial';
export const LoadingButtonState = 'loading';
export const talkTrackIdentifierProp = 'iid';
export const talkTrackType = 'talktrack';