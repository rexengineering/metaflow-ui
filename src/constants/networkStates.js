export const INITIAL = "INITIAL";
export const REQUEST = "REQUEST";
export const SUCCESS = "SUCCESS";
export const FAILURE = "FAILURE";

export default {
  INITIAL,
  REQUEST,
  SUCCESS,
  FAILURE
}
