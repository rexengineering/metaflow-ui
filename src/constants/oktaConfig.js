const config = {
  oidc: {
    clientId: window.REX.OKTA.CLIENT_ID,
    issuer: window.REX.OKTA.ISSUER,
    redirectUri: window.REX.OKTA.REDIRECT_URI,
    scopes: ["openid", "profile", "email"],
    pkce: true,
    disableHttpsCheck: window.REX.OKTA.OKTA_TESTING_DISABLEHTTPSCHECK,
  },
  resourceServer: {
    messagesUrl: "http://localhost:5000/api/messages",
  },
};

export default config;
