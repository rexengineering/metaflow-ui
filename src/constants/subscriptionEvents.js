export const START_WORKFLOW = "START_WORKFLOW";
export const UPDATE_WORKFLOW = "UPDATE_WORKFLOW";
export const FINISH_WORKFLOW = "FINISH_WORKFLOW";
export const START_TASK = "START_TASK";
export const UPDATE_TASK = "UPDATE_TASK";
export const FINISH_TASK = "FINISH_TASK";
export const START_BROADCAST = "START_BROADCAST";
export const FINISH_BROADCAST = "FINISH_BROADCAST";
export const ERROR_BROADCAST = "ERROR_BROADCAST";
export const KEEP_ALIVE_SUCCESS = "SUCCESS";
export const KEEP_ALIVE_FAILURE = "FAILURE";