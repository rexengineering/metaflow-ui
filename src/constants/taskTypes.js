import { string, number, boolean } from "yup";

// Inputs
export const TEXT = "TEXT";
export const PHONE_NUMBER = "PHONE_NUMBER";
export const PERCENTAGE = "PERCENTAGE";
export const CURRENCY = "CURRENCY";
export const BOOLEAN = "BOOLEAN";
export const INTEGER = "INTEGER";
export const FLOAT = "FLOAT";

export function isInputType(field) {
  return [
    TEXT,
    PHONE_NUMBER,
    PERCENTAGE,
    CURRENCY,
    BOOLEAN,
    INTEGER,
    FLOAT,
  ].includes(field);
}

// Info
export const TABLE = "TABLE";
export const COPY = "COPY";
export const WORKFLOW_INSTANTIATOR = "WORKFLOW";

export function isInfoType(type) {
  return [TABLE, COPY, WORKFLOW_INSTANTIATOR].includes(type);
}

const numberValidationMessage = "Please enter a number";

export const validationSchemaMapping = {
  TEXT: string("Please enter a value"),
  PHONE_NUMBER: string("Please enter a value").length(10),
  PERCENTAGE: number(numberValidationMessage).typeError(
    numberValidationMessage
  ),
  CURRENCY: number(numberValidationMessage).typeError(numberValidationMessage),
  BOOLEAN: boolean(),
  INTEGER: number(numberValidationMessage)
    .typeError(numberValidationMessage)
    .integer(),
  FLOAT: number(numberValidationMessage).typeError(numberValidationMessage),
};

const taskTypes = {
  TEXT,
  PHONE_NUMBER,
  PERCENTAGE,
  CURRENCY,
  BOOLEAN,
  INTEGER,
  FLOAT,
  TABLE,
  COPY,
  WORKFLOW_INSTANTIATOR,
};

export default taskTypes;
