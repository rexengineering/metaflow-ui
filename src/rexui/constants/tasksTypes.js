import { string, number, boolean } from "yup";


const numberValidationMessage = "Please enter a number";

export const TEXT = "TEXT";
export const PHONE_NUMBER = "PHONE_NUMBER";
export const PERCENTAGE = "PERCENTAGE";
export const CURRENCY = "CURRENCY";
export const BOOLEAN = "BOOLEAN";
export const INTEGER = "INTEGER";
export const FLOAT = "FLOAT";
export const TABLE = "TABLE";
export const COPY = "COPY";
export const WORKFLOW_INSTANTIATOR = "WORKFLOW";

export const validationSchemaMapping = {
    TEXT: string("Please enter a value"),
    PHONE_NUMBER: string("Please enter a value").length(10),
    PERCENTAGE: number(numberValidationMessage).typeError(numberValidationMessage),
    CURRENCY: number(numberValidationMessage).typeError(numberValidationMessage),
    BOOLEAN: boolean(),
    INTEGER: number(numberValidationMessage)
        .typeError(numberValidationMessage)
        .integer(),
    FLOAT: number(numberValidationMessage).typeError(numberValidationMessage),
};

export function isInfoType(type) {
    return [TABLE, COPY, WORKFLOW_INSTANTIATOR].includes(type);
}

export function isInputType(field) {
    return [
        TEXT,
        PHONE_NUMBER,
        PERCENTAGE,
        CURRENCY,
        BOOLEAN,
        INTEGER,
        FLOAT,
    ].includes(field);
}