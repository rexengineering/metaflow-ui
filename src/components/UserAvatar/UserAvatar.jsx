import React from "react";
import {Avatar, Badge, ButtonBase, makeStyles, useTheme} from "@material-ui/core";
import PropTypes from "prop-types";
import { BUSY, OFFLINE, ONLINE } from "../../constants";
import UserMenu from "./UserMenu";

const useStyles = makeStyles((theme) => ({
  container: {
    borderRadius: "50%",
  },
  avatar: {
    width: theme.spacing(6.25),
    height: theme.spacing(6.25),
    background: theme.palette.grey["400"],
  },
}));

function UserAvatar({
  currentStatus,
  username,
  userProfilePhotoURL,
  isDisabled,
  onUserAvatarClick,
  onStatusSelection,
  onClose,
  menuAnchor,
}) {
  const classes = useStyles();
  const { palette } = useTheme();
  const statuses = {
    [ONLINE]: palette.primary.main,
    [OFFLINE]: palette.background.default,
    [BUSY]: palette.miscellaneous1?.main,
  };
  const userNameAvatar = username?.charAt(0);
  return (
    <>
      <ButtonBase className={classes.container} onClick={onUserAvatarClick}>
        <Badge
          color={statuses[currentStatus]}
          badgeContent=" "
          variant="dot"
          invisible={isDisabled}
          overlap="circle"
        >
          <Avatar src={userProfilePhotoURL} className={classes.avatar}>
            {userNameAvatar}
          </Avatar>
        </Badge>
      </ButtonBase>
      <UserMenu
        status={currentStatus}
        statuses={statuses}
        menuAnchor={menuAnchor}
        onMenuClose={onClose}
        onStatusSelection={onStatusSelection}
      />
    </>
  );
}

UserAvatar.defaultProps = {
  userProfilePhotoURL: "",
  isDisabled: "",
  onUserAvatarClick: null,
  onStatusSelection: null,
  onClose: null,
  menuAnchor: null,
};

UserAvatar.propTypes = {
  currentStatus: PropTypes.string.isRequired,
  username: PropTypes.string.isRequired,
  userProfilePhotoURL: PropTypes.string,
  isDisabled: PropTypes.string,
  onUserAvatarClick: PropTypes.func,
  onStatusSelection: PropTypes.func,
  onClose: PropTypes.func,
  menuAnchor: PropTypes.node,
};

export default UserAvatar;
