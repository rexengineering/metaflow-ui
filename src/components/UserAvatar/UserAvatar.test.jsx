import { fireEvent, render, screen } from "@testing-library/react";
import React from "react";
import UserAvatar from "./UserAvatar";
import { ONLINE } from "../../constants";

describe("<UserAvatar />", () => {
  const userName = "Gabriel";
  const renderUserAvatar = (
    userProfilePictureURL = null,
    onClick = null,
    onStatusChange = null,
    disabled = "",
    onClose
  ) => (
    <UserAvatar
      currentStatus={ONLINE}
      username={userName}
      userProfilePhotoURL={userProfilePictureURL}
      isDisabled={disabled}
      onUserAvatarClick={onClick}
      onStatusSelection={onStatusChange}
      onClose={onClose}
      menuAnchor={React.createRef()}
    />
  );

  it("should render", () => {
    const { asFragment } = render(renderUserAvatar());
    const initialRender = asFragment();

    expect(initialRender).not.toBeNull();
  });

  it("should match snapshot", () => {
    const { asFragment } = render(renderUserAvatar());
    const initialRender = asFragment();

    expect(initialRender).toMatchSnapshot();
  });

  it("should show the initial letter of username when src is not provided", () => {
    render(renderUserAvatar());
    const initialLetter = userName.charAt(0);
    const initialLetterNode = screen.getByText(initialLetter);
    expect(initialLetterNode).toBeInTheDocument();
  });

  it("should show user profile picture when a src is provided", () => {
    render(
      renderUserAvatar(
        "https://hackernoon.com/hn-images/1*Y0UYuGcFGSCfs5Eexafq6A.png"
      )
    );
    const profilePicture = screen.getByRole("img", { hidden: true });
    expect(profilePicture).toBeInTheDocument();
  });

  it("should handle click event and status change event properly", () => {
    const handleClick = jest.fn();
    const status = "online";
    const handleStatusChange = jest.fn((newStatus) => {
      expect(newStatus === status).toBeTruthy();
    });
    const handleOnClose = jest.fn();

    render(
      renderUserAvatar(
        null,
        handleClick,
        handleStatusChange,
        undefined,
        handleOnClose
      )
    );

    const avatar = screen.getByRole("button", { hidden: true });
    expect(avatar).toBeInTheDocument();
    fireEvent.click(avatar);
    expect(handleClick).toHaveBeenCalledTimes(1);
  });

/*
 it("should handle click event when disabled properly", () => {
    const handleClick = jest.fn();
    render(renderUserAvatar(null, handleClick, null, "true"));

    const avatar = screen.getByRole("button");
    expect(avatar).toBeInTheDocument();
    fireEvent.click(avatar);
    expect(handleClick).toHaveBeenCalledTimes(0);
  });*/
});
