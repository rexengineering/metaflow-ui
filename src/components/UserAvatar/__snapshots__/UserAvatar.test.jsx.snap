// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserAvatar /> should match snapshot 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root makeStyles-container-7"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiBadge-root"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-8 MuiAvatar-colorDefault"
      >
        G
      </div>
      <span
        class="MuiBadge-badge MuiBadge-anchorOriginTopRightCircle MuiBadge-dot"
      />
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
